src/v/cluster/topics_frontend.h

```diff
@@ -228,6 +228,8 @@ class topics_frontend {
     /// Trigger shard placement rebalancing for partitions in this node.
     ss::future<errc> trigger_local_partition_shard_rebalance();
 
+    const metadata_cache& get_metadata_cache() const { return _metadata_cache; }
```

Comment:
nit: i know it is convenient but should this method be in a public interface of `topics_frontend` ? maybe it would be better to pass in the metadata cache directly into `datalake_coordinator` instead the `topic_frontend` it hass all necessarry apis